iam_pipes.tf

```diff
@@ -0,0 +1,396 @@
+      actions = [
+        "states:StartExecution"
+      ]
+    }
+
+    api_gateway = {
+      actions = [
+        "execute-api:Invoke"
+      ]
+    }
+
+    api_destination = {
+      actions = [
+        "events:InvokeApiDestination"
+      ]
+    }
+
+    batch = {
+      actions = [
+        "batch:SubmitJob"
+      ]
+    }
+
+    logs = {
+      actions = [
+        "logs:DescribeLogGroups",
+        "logs:DescribeLogStreams",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ]
+    }
+
+    ecs = {
+      actions = [
+        "ecs:RunTask"
+      ]
+    }
+
+    ecs_iam_passrole = {
+      actions = [
+        "iam:PassRole"
+      ]
+    }
+
+    eventbridge = {
+      actions = [
+        "events:PutEvents"
+      ]
+    }
+
+    firehose = {
+      actions = [
+        "firehose:PutRecord"
+      ]
+    }
+
+    inspector = {
+      actions = [
+        "inspector:CreateAssessmentTemplate" # ???
+      ]
+    }
+
+    redshift = {
+      actions = [
+        "redshift-data:ExecuteStatement" # ???
+      ]
+    }
+
+    sagemaker = {
+      actions = [
+        "sagemaker:CreatePipeline" # ???
+      ]
+    }
+
+    sns = {
+      actions = [
+        "sns:Publish"
+      ]
+    }
+  }
+}
+
+#################################
+# IAM role for EventBridge Pipes
+#################################
+
+data "aws_iam_policy_document" "assume_role_pipe" {
+  for_each = local.role_for_pipes
+
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["pipes.amazonaws.com"]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceAccount"
+      values   = [data.aws_caller_identity.current.account_id]
+    }
+
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:pipes:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:pipe/${each.value.Name}"]
+    }
+  }
+}
+
+resource "aws_iam_role" "eventbridge_pipe" {
+  for_each = local.role_for_pipes
+
+  name                  = each.value.role_name
+  description           = try(each.value.role_description, null)
+  path                  = try(each.value.role_path, null)
+  force_detach_policies = try(each.value.role_force_detach_policies, null)
+  permissions_boundary  = try(each.value.role_permissions_boundary, null)
+  assume_role_policy    = data.aws_iam_policy_document.assume_role_pipe[each.key].json
+
+  tags = merge({ Name = each.value.role_name }, var.tags)
+}
+
+
+##############################
+# Predefined service policies
+##############################
+
+data "aws_iam_policy_document" "service" {
+  for_each = { for k, v in local.service_integrations_for_pipes : k => v if try(v.attach_policies_for_integrations, true) }
+
+  dynamic "statement" {
+    for_each = { for s_k, s_v in each.value : s_k => s_v if length(compact(s_v)) > 0 }
+
+    content {
```

Comment:
I'm gonna take your word on all of this 😅

Reply:
Trust me - it works on my machine! 🤣